Description: upstream patch from upcoming 7.0.36 to allow precompilation 
 of JSPs; TagPluginManager.init() throws NPE when used in Ant <jasper> task
Bug: https://issues.apache.org/bugzilla/show_bug.cgi?id=54440
Origin: https://issues.apache.org/bugzilla/attachment.cgi?id=29860
Forwarded: not-needed

--- a/java/org/apache/jasper/JspC.java
+++ b/java/org/apache/jasper/JspC.java
@@ -1516,7 +1516,7 @@
         URL urlsA[]=new URL[urls.size()];
         urls.toArray(urlsA);
         loader = new URLClassLoader(urlsA, this.getClass().getClassLoader());
-
+        context.setClassLoader(loader);
     }
 
     /**
--- a/java/org/apache/jasper/servlet/JspCServletContext.java
+++ b/java/org/apache/jasper/servlet/JspCServletContext.java
@@ -77,6 +77,11 @@
      */
     protected URL myResourceBaseURL;
 
+
+    /**
+     * Class Loader
+     */
+    private ClassLoader loader;
 
     // ----------------------------------------------------------- Constructors
 
@@ -612,7 +617,11 @@
 
     @Override
     public ClassLoader getClassLoader() {
-        return null;
+        return loader;
+    }
+
+    public void setClassLoader(ClassLoader loader) {
+    	this.loader = loader;
     }
 
 
